### PR TITLE
Added shift parameter to TypeDate

### PR DESCRIPTION
**Pull request by sol-0**
# Contribution checklist

## Brief description
Added `withShift` parameter to `TypeDate` so that it can be passed in model property decorator. This parameter can be used to show/hide time shift menu.

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [x] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

**Review by BykhovDenis — APPROVED** [review]

File: plugins/view-resources/src/components/ObjectPresenter.svelte [hunk 1]
```diff
@@ -50,6 +50,7 @@
 
   $: if (doc !== undefined) {
     getObjectPresenter(client, doc._class, { key: '' }).then((p) => {
+      console.log(`PRESENTER: ${p._class}, ${p.presenter.name}`)
```
Comment (haiodo) [inline plugins/view-resources/src/components/ObjectPresenter.svelte]: I do not think we need this logged.

File: packages/presentation/src/components/AttributeBarEditor.svelte [hunk 2]
```diff
@@ -56,9 +56,11 @@
 
     const typeClass = hierarchy.getClass(presenterClass.attrClass)
     const editorMixin = hierarchy.as(typeClass, mixinRef)
+    console.log(`presenterClass: ${JSON.stringify(presenterClass)}, typeClass: ${JSON.stringify(typeClass)}, _class: ${typeClass._class}, _id: ${typeClass._id}, editorMixin: ${JSON.stringify(editorMixin)}, mixinRef: ${mixinRef}`)
```
Comment (haiodo) [inline packages/presentation/src/components/AttributeBarEditor.svelte]: I do not think this need to be logged.

File: packages/presentation/src/components/AttributeBarEditor.svelte [hunk 3]
```diff
@@ -56,9 +56,11 @@
...
     if (category === 'array' && editorMixin.inlineEditor === undefined) {
       return
     }
+    console.log(`editorMixin.inlineEditor: ${JSON.stringify(editorMixin.inlineEditor)}`)
```
Comment (haiodo) [inline packages/presentation/src/components/AttributeBarEditor.svelte]: I do not think this need to be logged.

File: plugins/view-resources/src/components/ObjectPresenter.svelte (hunk 1 above)
Comment (sol-0) [reply plugins/view-resources/src/components/ObjectPresenter.svelte]: Sorry, I forgot to remove debugging logs. Removed it now, thanks

File: packages/presentation/src/components/AttributeBarEditor.svelte (hunk 2 above)
Comment (sol-0) [reply packages/presentation/src/components/AttributeBarEditor.svelte]: Removed, thanks

File: packages/presentation/src/components/AttributeBarEditor.svelte (hunk 3 above)
Comment (sol-0) [reply packages/presentation/src/components/AttributeBarEditor.svelte]: Removed, thanks

**Review by haiodo — APPROVED** [review]

**Pull request merged**